src/Microsoft.DotNet.Interactive.Tests/StdioConnectionTests.cs

```diff
@@ -0,0 +1,158 @@
+
+            // Copy each file into the new directory.
+            foreach (FileInfo fi in source.GetFiles())
+            {
+                Console.WriteLine(@"Copying {0}\{1}", target.FullName, fi.Name);
+                fi.CopyTo(Path.Combine(target.FullName, fi.Name), true);
+            }
+
+            // Copy each subdirectory using recursion.
+            foreach (DirectoryInfo diSourceSubDir in source.GetDirectories())
+            {
+                DirectoryInfo nextTargetSubDir =
+                    target.CreateSubdirectory(diSourceSubDir.Name);
+                CopyAll(diSourceSubDir, nextTargetSubDir);
+            }
+        }
+        RegisterForDisposal(Disposable.Create(() =>
+        {
+            dstDir.Delete(true);
+        }));
+        return new FileInfo(Path.Combine(dstDir.FullName, toolFileInfo.Name));
+    }
+
+    [Fact]
+    public async Task stdio_server_encoding_is_utf_8()
+    {
+        var configuration = CreateConnectionConfiguration();
+
+        await CreateRemoteKernelTopologyAsync(configuration);
+
+        using var localCompositeKernel = new CompositeKernel
+        {
+            new FakeKernel("fsharp")
+        };
+
+        ConfigureConnectCommand(localCompositeKernel);
+
+        localCompositeKernel.DefaultKernelName = "fsharp";
+
+        var connectToRemoteKernel = CreateConnectionCommand(configuration);
+
+        await localCompositeKernel.SendAsync(connectToRemoteKernel);
+
+        var res = await localCompositeKernel.SendAsync(new SubmitCode("System.Console.InputEncoding.EncodingName + \"/\" + System.Console.OutputEncoding.EncodingName", "newKernelName"));
+        var expected = Encoding.UTF8.EncodingName + "/" + Encoding.UTF8.EncodingName;
+
+        var events = res.KernelEvents.ToSubscribedList();
+
+        events
+            .Should()
+            .EventuallyContainSingle<DisplayEvent>(
+                where: d => d.FormattedValues.Any(FormattedValue => FormattedValue.Value == expected),
+                timeout: 10_000);
+    }
+
+    protected override SubmitCode CreateConnectionCommand(StdioConnectionTestConfiguration configuration)
+    {
+        return new SubmitCode(
+            $"#!connect stdio --kernel-name newKernelName --command \"{configuration.Command}\" {string.Join(" ", configuration.Args)}");
+    }
+
+    protected override void ConfigureConnectCommand(CompositeKernel compositeKernel)
+    {
+        compositeKernel.UseKernelClientConnection(new ConnectStdIoCommand());
+    }
+
+    protected override Task<IDisposable> CreateRemoteKernelTopologyAsync(StdioConnectionTestConfiguration configuration)
+    {
+        return Task.FromResult( Disposable.Empty);
+    }
+
+    protected override Task<IDisposable> ConnectHostAsync(CompositeKernel remoteKernel, StdioConnectionTestConfiguration configuration)
+    {
+        return Task.FromResult(Disposable.Empty);
+    }
+}
+
+public class StdioConnectionTestConfiguration
+{
+    public string Command { get; set; }
```

Comment:
This name is ambiguous. Maybe ExePath? Or, realistically in this context, this is `DotNetExePath`? Is it ever different? Is it meaningful to make this settable?

Reply:
That is the binary to run, trying to model like the process startinfo thing